iwyu-check-license-header.py

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env python
+
+##===---------- iwyu-check-license-header.py - check license headers ------===##
+#
+#                     The LLVM Compiler Infrastructure
+#
+# This file is distributed under the University of Illinois Open Source
+# License. See LICENSE.TXT for details.
+#
+##===----------------------------------------------------------------------===##
+
+from __future__ import print_function
+import sys
+import os
+import re
+import argparse
+
+
+HDRFORMAT = """
+{c}
+{c}                     The LLVM Compiler Infrastructure
+{c}
+{c} This file is distributed under the University of Illinois Open Source
+{c} License. See LICENSE.TXT for details.
+{c}
+{t}===----------------------------------------------------------------------==={t}
+"""
+
+
+def make_hdrformat(one, two):
+    """ Materialize HDRFORMAT based on the one and two comment styles.
+
+    The one and two args should be '#' and '##' or '//' and '//', respectively.
+    Returns a list of lines.
+    """
+    r = HDRFORMAT.lstrip().format(c=one, t=two)
+    return r.splitlines()
+
+
+def make_license_header(filename, one, two):
+    """ Build a valid license header from filename and comment styles.
+
+    The one and two args should be '#' and '##' or '//' and '//', respectively.
+    Returns a list of lines.
+    """
+    assert len(two) == 2
+    filename = os.path.basename(filename)
+
+    def dashes():
+        c = 80
+        c -= 2  # take off leading ##
+        c -= 6  # take off leading ===---
+        c -= len(filename) + 2  # take off filename with a space on either side
+        c -= 3  # take off trailing ===
+        c -= 2  # take off trailing ##
+        return '-' * c
+
+    firstline = '%s===--- %s %s===%s' % (two, filename, dashes(), two)
```

Comment:
If the filename is longer than 65 chars (80-2-6-2-3-2) this will fail.
Maybe better to calc the dashes count and truncate the filename by that amount in that case?

Reply:
Fixed.